plugin-server/src/worker/ingestion/app-metrics.ts

```diff
@@ -88,9 +103,6 @@ export class AppMetrics {
         // However, we also don't want to wait too long, nor have the queue grow too big resulting in
         // the flush taking a long time.
         const now = Date.now()
-        if (now - this.lastFlushTime > this.flushFrequencyMs || this.queueSize > this.maxQueueSize) {
-            await this.flush()
-        }
```

Comment:
This was moved down to the end of this method to ensure that this block would reliably be entered during the test suite.